sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -321,6 +321,12 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case CreateTableUsing(tableName, userSpecifiedSchema, provider, false, options) =>
         sys.error("Tables created with SQLContext must be TEMPORARY. Use a HiveContext instead.")
 
+      case DropTable(table, isExists, temporary) =>
+        if (!temporary) {
+          sys.error("Tables dropped with SQLContext must be TEMPORARY, Use a HiveContext instead.")
+        }
+        ExecutedCommand(DropTempTable(table, isExists, temporary)) :: Nil
+
```

Comment:
Actually, you don't need `DropTable` or this `case` branch. I'd suggesting:

1. moving the `if` into `DropTempTable`,
1. removing `DropTable`
1. replacing `DropTable` in the SQL parser with `DropTempTable`
1. renaming `DropTempTable` to `DropTable`